src/main/java/org/zalando/failsafeactuator/aspect/FailsafeBreakerAspect.java

```diff
@@ -1,59 +1,75 @@
+ * sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * <p>The above copyright notice and this permission notice shall be included in all copies or
+ * substantial portions of the Software.
+ *
+ * <p>THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING
+ * BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
  */
 package org.zalando.failsafeactuator.aspect;
 
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.concurrent.Callable;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import net.jodah.failsafe.CircuitBreaker;
 import net.jodah.failsafe.Failsafe;
 import org.aopalliance.intercept.MethodInterceptor;
 import org.aopalliance.intercept.MethodInvocation;
+import org.springframework.boot.actuate.metrics.CounterService;
+import org.zalando.failsafeactuator.metrics.CountingListener;
 import org.zalando.failsafeactuator.service.CircuitBreakerRegistry;
 
-import java.lang.reflect.InvocationTargetException;
-import java.lang.reflect.Method;
-import java.util.concurrent.Callable;
-
-
 @Slf4j
 @RequiredArgsConstructor
 public class FailsafeBreakerAspect implements MethodInterceptor {
 
   private final CircuitBreakerRegistry circuitBreakerRegistry;
+  private final CounterService counterService;
 
+  @Override
   public Object invoke(final MethodInvocation methodInvocation) throws Throwable {
     final Method method = methodInvocation.getMethod();
-    final org.zalando.failsafeactuator.aspect.Failsafe breaker = method.getAnnotation(org.zalando.failsafeactuator.aspect.Failsafe.class);
+    final org.zalando.failsafeactuator.aspect.Failsafe breaker =
+        method.getAnnotation(org.zalando.failsafeactuator.aspect.Failsafe.class);
+
 
     final CircuitBreaker circuitBreaker = circuitBreakerRegistry.getOrCreate(breaker.value());
     try {
       return Failsafe.with(circuitBreaker)
-                     .get(new Callable<Object>() {
-                       @Override
-                       public Object call() throws Exception {
-                         try {
-                           return methodInvocation.proceed();
-                         } catch (Throwable throwable) {
-                           throw new FailsafeBreakerWrappedException(throwable);
-                         }
-                       }
-                     });
-    } catch (RuntimeException e) {
+          .with(new CountingListener<>(counterService, breaker.value()))
+          .get(
```

Comment:
This won't work for methods that return a Future.

Reply:
We should file a new Issue for that since this logic wasn't introduced by this PR.

Reply:
See #58

Reply:
Hey @whiskeysierra now as I am looking into it, it's not really clear to me what you mean here. Could maybe explain a bit more in detail or extend the issue? Would be awesome!

Reply:
There are two kinds of methods, sync and async. `get` is for synchronous methods. For something that returns a future, you should use `future`: https://github.com/jhalterman/failsafe#completablefuture-integration

Let's say you have a method that performs a remote request asynchronously and returns a `Future<User>`.

- `get` would only retry the synchronous part of that, i.e. if anything failed during the creation of the future, but not during
- `future` would be aware that the operation is asynchronous and it would correctly retry the asynchronous operation